src/sentry/relay/config/__init__.py

```diff
@@ -162,6 +163,35 @@ def get_dynamic_sampling_config(project: Project) -> Optional[Mapping[str, Any]]
     return None
 
 
+class TransactionNameRule(TypedDict):
+    pattern: str
+    expiry: str
+    scope: Mapping[str, str]
+    redaction: Mapping[str, str]
+
+
+def get_transaction_names_config(project: Project) -> Optional[Sequence[TransactionNameRule]]:
+    if not features.has("organizations:transaction-name-sanitization", project.organization):
+        return None
+
+    cluster_rules = get_sorted_rules(project)
+    if cluster_rules is None or len(cluster_rules) == 0:
+        return None
+
+    return [_get_tx_name_rule(p, e) for p, e in cluster_rules]
+
+
+def _get_tx_name_rule(pattern: str, expiry: int) -> TransactionNameRule:
+    return TransactionNameRule(
+        pattern=pattern,
+        expiry=datetime.fromtimestamp(expiry, tz=timezone.utc).isoformat(),
```

Comment:
We now store the `last_seen` date instead of the expiry date, so we should add some extra time here (90 days?).

```suggestion
        expiry=datetime.fromtimestamp(expiry, tz=timezone.utc).isoformat() + SOME_TTL,
```